### PR TITLE
Support ADFS encrypted key

### PR DESCRIPTION
**Pull request by joshwetzel**
ADFS does not use the `ds` namespace on their `KeyInfo` node.

### CODE REVIEW

**Comment by pitbulk** [comment]
Thanks for sharing that code, I will review it soon.

**Comment by joshwetzel** [comment]
@pitbulk have you had a chance to review this?

**Comment by pitbulk** [comment]
Yes I reviewed it and I'm not a fan of ignore namespaces.
In [php-saml](https://github.com/onelogin/php-saml/blob/master/lib/Saml2/Response.php#L1072) I did a workaround to fix an issue when the decrypted data has issue with namespace.

Trying to figure out something here. Maybe, meanwhile can you provide the encoded SAMLResponse encryped with non production keys (and provide me by mail the necessary data to debug)?

**Comment by joshwetzel** [comment]
Perhaps something not so heavy handed? After looking at this again, I think we only need to update the `EncryptedKey` path.

```diff
diff --git a/lib/onelogin/ruby-saml/utils.rb b/lib/onelogin/ruby-saml/utils.rb
index dd0c1b7..a0a9850 100644
--- a/lib/onelogin/ruby-saml/utils.rb
+++ b/lib/onelogin/ruby-saml/utils.rb
@@ -131,10 +131,9 @@ module OneLogin
       def self.retrieve_symmetric_key(encrypt_data, private_key)
         encrypted_key = REXML::XPath.first(
           encrypt_data,
-          "./ds:KeyInfo/xenc:EncryptedKey or \
-           //xenc:EncryptedKey[@Id=$id]",
-          { "ds" => DSIG, "xenc" => XENC },
-          { "id" =>  self.retrieve_symetric_key_reference(encrypt_data) }
+          "./ds:KeyInfo/xenc:EncryptedKey | ./KeyInfo/e:EncryptedKey | //xenc:EncryptedKey[@Id=$id]",
+          { "ds" => DSIG, "xenc" => XENC, "e" => XENC },
+          { "id" => self.retrieve_symetric_key_reference(encrypt_data) }
         )
 
         encrypted_symmetric_key_element = REXML::XPath.first(
```

**Comment by joshwetzel** [comment]
@pitbulk let me know what you think of the latest changes I pushed up.

File: lib/onelogin/ruby-saml/utils.rb [hunk 1]
```diff
@@ -139,23 +139,25 @@ def self.decrypt_data(encrypted_node, private_key)
       def self.retrieve_symmetric_key(encrypt_data, private_key)
         encrypted_key = REXML::XPath.first(
           encrypt_data,
-          "./ds:KeyInfo/xenc:EncryptedKey or \
-           //xenc:EncryptedKey[@Id=$id]",
-          { "ds" => DSIG, "xenc" => XENC },
-          { "id" =>  self.retrieve_symetric_key_reference(encrypt_data) }
+          "./ds:KeyInfo/xenc:EncryptedKey | ./KeyInfo/e:EncryptedKey | //xenc:EncryptedKey[@Id=$id]",
+          { "ds" => DSIG, "xenc" => XENC, "e" => XENC },
```
Comment (pitbulk) [inline lib/onelogin/ruby-saml/utils.rb]: why add "e" => XENC if already was defined "xenc" => XENC? It should be:

"./ds:KeyInfo/xenc:EncryptedKey | ./KeyInfo/xenc:EncryptedKey | //xenc:EncryptedKey[@Id=$id]",
{ "ds" => DSIG, "xenc" => XENC},

Comment (joshwetzel) [inline lib/onelogin/ruby-saml/utils.rb]: Thanks, updated.

**Comment by pitbulk** [comment]
I will release a new version soon.

**Pull request merged**